feat: Add new Pathfinder 2e Creature layout (#436)

miscoined opened the pull request:
## Pull Request Description

Add a new default Pathfinder 2e Creature layout. See #436 for rationale and background. This adds this as a new layout, without replacing the existing Basic layout. I'm opening this as a draft PR for now until I finish adding documentation and layout restoration functionality - I thought early feedback would be useful in this case.

## Changes Proposed

- [x] Add a new Pathfinder 2e Creature Layout
- [x] Update Pathfinder 2e CSS from the upstream repo
- [x] Add documentation for the new layout
- [x] Add/update the default layout block for layout restoration functionality

## Related Issues

Progresses #436.

## Checklist (WIP)

- [x] I have read the contribution guidelines and code of conduct.
- [x] I have tested the changes locally and they are working as expected.
- [x] I have added appropriate comments and documentation for the code changes.
- [x] My code follows the coding style and standards of this project.
- [x] I have rebased my branch on the latest main (or master) branch.
- [x] All tests (if applicable) have passed successfully.
- [x] I have run linters and fixed any issues.
- [x] I have checked for any potential security issues or vulnerabilities.

## Screenshots (if applicable)

Will add later, before flipping this out of a draft PR

valentine195 commented:
Hey @miscoined, this looks great. Is this meant to slot into existing PF2e creature stat blocks used by the TTRPG share?

valentine195 commented on src/settings/layout/blocks/ui/block.ts:
```diff
@@ -926,6 +979,17 @@ class TableModal extends BasicModal<TableItem> {
                 });
             });
+        new Setting(el)
+            .setName("Do Not Add Property as CSS Class")
```
I usually try to prefer making these things _perform_ the task rather than turning it off. Could you please refactor this setting to that?

miscoined replied:
I did that for consistency with the same option for other blocks. I'm happy to change it if you'd prefer the inconsistency. I'd also be happy to change the other options as well (I'd probably do it in a separate PR), but I'm not sure if that would mean that existing user's settings would actually end up inverted. Just checking what you'd prefer :)

valentine195 commented on src/layouts/pathfinder 2e/pf2e.ts:
```diff
@@ -471,3 +355,457 @@ export const LayoutPF2e: DefaultLayout = {
+                                    ]
+                                }
+                            ],
+                            condition: "return plugin.app.plugins.enabledPlugins.has(\"initiative-tracker\");\n"
```
This is actually genius! Very clever.

valentine195 commented on src/layouts/pathfinder 2e/pf2e.ts:
```diff
@@ -471,3 +355,457 @@ export const LayoutPF2e: DefaultLayout = {
+                    markdown: true,
+                    headingProp: false,
+                    heading: "",
+                    callback: "var s = property.bonus < 0 ? \"-\" : \"+\";\ns += Math.abs(property.bonus);\nif (property.desc) {\n  s += \" \" + property.desc\n}\nif (property.damage) {\n  s += \", __Damage__ \" + property.damage\n}\nreturn s"
```
This will return `NaN` if `property.bonus` is undefined

miscoined replied:
Done

valentine195 commented:
Sorry I let this sit for so long; been a very busy summer. I appreciate the work you have been doing.

miscoined commented:
Hey! I had a busy while myself so I'm sorry it took so long to get back to this. I've fixed up any merge conflicts, updated the documentation, and made the changes you requested. I've been using this on my own personal vault for a while now and haven't noticed any problems.

valentine195 commented:
I added you to the repo but you are under no obligation to do anything :) I just had a kid so I’ve been out of commission for awhile.

miscoined commented:
> I added you to the repo but you are under no obligation to do anything :) I just had a kid so I’ve been out of commission for awhile.

Thank you for the quick merge and the add! and congrats on the new kid :D
